go/private/actions/archive.bzl

```diff
@@ -129,12 +138,28 @@ def emit_archive(go, source = None, _recompile_suffix = "", recompile_internal_d
             archives = direct,
             out_lib = out_lib,
             out_export = out_export,
+            out_nogo_log = out_nogo_log,
             gc_goopts = source.gc_goopts,
             cgo = False,
             testfilter = testfilter,
             recompile_internal_deps = recompile_internal_deps,
         )
 
+    if go.nogo:
+        validations.append(out_nogo_validation)
+        go.actions.run_shell(
```

Comment:
We try to avoid relying on `run_shell` on Windows. Can you think of a way to replace this with a `run`? 

For example, could this logic be moved into https://github.com/bazelbuild/rules_go/blob/master/go/tools/builders/generate_nogo_main.go?